src/modules/75-cf/components/FlagActivation/__tests__/JiraIssueList.test.tsx

```diff
@@ -48,6 +78,156 @@ describe('JiraIssueList', () => {
+    expect(screen.getByText('add')).toBeInTheDocument()
+    expect(screen.getByText('cancel')).toBeInTheDocument()
+  })
+
+  test('it should let the user save a Jira Issue correctly', async () => {
+    renderComponent()
+
+    const jiraModalButton = screen.getByRole('button', { name: 'cf.featureFlags.jira.newJiraIssueButton' })
+    expect(jiraModalButton).toBeInTheDocument()
+    userEvent.click(jiraModalButton)
+
+    // type into issue search field
+    expect(screen.getByPlaceholderText('- cf.featureFlags.jira.inputPlaceholder -')).toBeInTheDocument()
+    userEvent.click(screen.getByPlaceholderText('- cf.featureFlags.jira.inputPlaceholder -'))
+    await userEvent.type(screen.getByPlaceholderText('- cf.featureFlags.jira.inputPlaceholder -'), 'FD-7', {
+      allAtOnce: true
+    })
+
+    // select issue from dropdown
+    await waitFor(() => expect(screen.getAllByRole('listitem')).toHaveLength(1))
+    expect(screen.getByRole('listitem')).toHaveTextContent('FD-7')
+    userEvent.click(screen.getByRole('listitem'))
+    await waitFor(() =>
+      expect(screen.getByPlaceholderText('- cf.featureFlags.jira.inputPlaceholder -')).toHaveValue('FD-7')
+    )
+
+    // submit form
+    userEvent.click(screen.getByText('add'))
+    await waitFor(() =>
+      expect(patchMutateMock).toHaveBeenCalledWith({
+        instructions: [
+          {
+            kind: 'addJiraIssueToFlag',
+            parameters: {
+              issueKey: 'FD-7'
+            }
+          }
+        ]
+      })
+    )
+  })
+
+  test('it should handle/show error on add button click', async () => {
+    renderComponent()
+
+    jest.spyOn(cfServiceMock, 'usePatchFeature').mockReturnValue({
+      loading: false,
+      mutate: patchMutateMock.mockRejectedValueOnce(false)
+    } as any)
+
+    const jiraModalButton = screen.getByRole('button', { name: 'cf.featureFlags.jira.newJiraIssueButton' })
+    expect(jiraModalButton).toBeInTheDocument()
+    userEvent.click(jiraModalButton)
+
+    // type into issue search field
+    expect(screen.getByPlaceholderText('- cf.featureFlags.jira.inputPlaceholder -')).toBeInTheDocument()
+    userEvent.click(screen.getByPlaceholderText('- cf.featureFlags.jira.inputPlaceholder -'))
+    await userEvent.type(screen.getByPlaceholderText('- cf.featureFlags.jira.inputPlaceholder -'), 'FD-7', {
+      allAtOnce: true
+    })
+
+    // select issue from dropdown
+    await waitFor(() => expect(screen.getAllByRole('listitem')).toHaveLength(1))
+    expect(screen.getByRole('listitem')).toHaveTextContent('FD-7')
+    userEvent.click(screen.getByRole('listitem'))
+    await waitFor(() =>
+      expect(screen.getByPlaceholderText('- cf.featureFlags.jira.inputPlaceholder -')).toHaveValue('FD-7')
+    )
+
+    // submit form
+    userEvent.click(screen.getByText('add'))
+    await waitFor(() => expect(screen.getByText('error')).toBeInTheDocument())
```

Comment:
100% not a blocker, just FYI that `findBy` combines `getBy` and `waitFor`, which is super handy for these kinds of things.
```suggestion
    expect(await screen.findByText('error')).toBeInTheDocument()
```

Reply:
good to know for future

Reply:
good to know for future